### ISSUE TITLE
Evaluator.php triggers PHP type error

### ISSUE THREAD

**Issue by SteveJobzniak** [mossadal/math-parser#5]
Hi! I found this bug when I wanted to try to trigger various exceptions by writing bad equations.

```
$ php -v
PHP 7.0.15 (cli) (built: Feb 10 2017 16:55:27) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2017 Zend Technologies
```

```php
        $equation = '2poww(2)';

        $parser = new \MathParser\StdMathParser();

        // Generate an abstract syntax tree.
        $AST = $parser->parse( $equation );

        // Evaluate the expression in the AST.
        $evaluator = new \MathParser\Interpreting\Evaluator();
        $value = $AST->accept( $evaluator );
```

```
PHP Warning:  array_key_exists() expects parameter 2 to be array, null given in ./vendor/mossadal/math-parser/src/MathParser/Interpreting/Evaluator.php on line 162

Warning: array_key_exists() expects parameter 2 to be array, null given in ./vendor/mossadal/math-parser/src/MathParser/Interpreting/Evaluator.php on line 162
```

**Comment by SteveJobzniak** [comment]
By the way, `"1+.1"` triggers an exception saying `Unknown token 0 encountered`.

Maybe add some parsing logic to say "if period, assume a zero to the left"? So that `"1+.1"` is treated as `"1+0.1"`?

**Comment by mossadal** [comment]
Thank you, this is fixed in 1.3.9. The default list of variables was set to `null` rather than `[]`.
